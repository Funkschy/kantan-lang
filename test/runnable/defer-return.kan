import "io";

def f(i: i32) {
    defer io.printf("%d\n", i);
    if i % 2 == 0 {
        defer io.printf("even\n");
        return;
    }

    io.printf("end\n");
}

def main() {
    f(1);
    f(2);
}
