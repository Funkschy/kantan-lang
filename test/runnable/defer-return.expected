end
1
even
2
